### PR TITLE
Add flake8 to the tidy process for Python files

### LINKED ISSUE #6236: Run flake8 on Python files as part of the tidy process
Would help find things like unused variables, unused imports, etc.

If someone could point me at where to include the `flake8` requirement, I'd be happy to make all the necessary changes to make flake8 happy

Initially at least, I imagine we can ignore most of the whitespace and line-length errors

#### Backstory:

https://github.com/servo/servo/pull/6235#issuecomment-107256558

https://github.com/servo/servo/pull/6235#issuecomment-107257052

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes #6236

Also included in this commit are the changes need to make flake8 pass
for the existing python file

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6279)

### CODE REVIEW

File: etc/servo_gdb.py [hunk 1]
```diff
@@ -7,19 +7,22 @@
 # option. This file may not be copied, modified, or distributed
 # except according to those terms.
 
-# A set of simple pretty printers for gdb to make debugging Servo a bit easier.
+"""
```
Comment (frewsxcv) [inline etc/servo_gdb.py]: To those unfamiliar with Python, this is a module level docstring, which seems appropriate considering the contents. Flake8 did not like the `#texthere` conventions for comments without a space inbetween the text and the `#` so this worked out

File: python/tidy.py [hunk 2]
```diff
@@ -7,17 +7,17 @@
...
 import fnmatch
 import itertools
 import re
+import sys
 from licenseck import licenses
 
 filetypes_to_check = [".rs", ".rc", ".cpp", ".c", ".h", ".py"]
 reftest_directories = ["tests/ref"]
 reftest_filetype = ".list"
+dependencies = ["./python/flake8-2.4.1-py2.py3-none-any.whl"]
```
Comment (frewsxcv) [inline python/tidy.py]: If anyone has a better suggestion for the location of this wheel, let me know. I just kind of threw it in there thinking someone might have a better idea where it should live

Comment (frewsxcv) [inline python/tidy.py]: @SimonSapin FYI, adding a wheel to the `sys.path` apparently works

**Comment by frewsxcv** [comment]
Travis:

```
ImportError: No module named pep8
```

Dependency issues, woo!

**Comment by frewsxcv** [comment]
Okay, this should be ready for review. I will squash before merging

**Comment by mbrubeck** [comment]
Review status: all files reviewed, 1 unresolved discussion, all commit checks successful.
Reviewed files:
- etc/servo_gdb.py @ r1
- python/dependencies/flake8-2.4.1-py2.py3-none-any.whl @ r2
- python/dependencies/pep8-1.6.2-py2.py3-none-any.whl @ r2
- python/dependencies/pyflakes-0.9.0-py2.py3-none-any.whl @ r2
- python/licenseck.py @ r1
- python/mach_bootstrap.py @ r1
- python/servo/bootstrap_commands.py @ r1
- python/servo/build_commands.py @ r1
- python/servo/command_base.py @ r1
- python/servo/devenv_commands.py @ r1
- python/servo/post_build_commands.py @ r3
- python/servo/testing_commands.py @ r1
- python/tidy.py @ r2
- tests/jquery/run_jquery.py @ r1
- tests/power/PowerMeasure.py @ r1

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6279)

**Comment by mbrubeck** [comment]
r=mbrubeck.  I was a little uncertain about including the dependencies as wheel files, but it seems reasonable in this case.  Verified that all the code is Expat/MIT licensed.

**Comment by frewsxcv** [comment]
Squashed

**Comment by frewsxcv** [comment]
Hm. That looks to be a valid issue homubors found. I will check it out in the morning.

**Comment by jdm** [comment]
I presume it's the change at https://github.com/servo/servo/pull/6279/files#diff-a139f7c2483c6cf4f78098b575503c04L297 causing problems?

**Comment by frewsxcv** [comment]
Yeah, I assume so, was going to fiddle around with that stuff later

**Comment by frewsxcv** [comment]
I think I just need to add in both of those imports, despite them not being explicitly used anywhere in that module. Let me know how that last force push looks

**Comment by frewsxcv** [comment]
Grr, same error

**Comment by frewsxcv** [comment]
I am just going to not touch that section since it appears to be so fragile

**Comment by frewsxcv** [comment]
Should be ready for review again. I reverted my `import` changes in `testing_commands.py`, I also downgraded `pep8` to a different version that is compatible with flake8.

**Pull request merged**